### PR TITLE
EN-62919: write build to rms

### PR DESCRIPTION
**Pull request by sarah-salinas**
This commit will test https://github.com/socrata/pipelines/pull/142, which creates vars for Release Metatdata Service integration and then it will add the integration to write to RMS for release builds.

### CODE REVIEW

**Comment by sarah-salinas** [comment]
## Testing

### Simulated Release Build
https://jenkins-build.socrata.com/job/soql-postgres-adapter-branch/49/console - 🟢 
--> I deleted the release tag created "soql-server-pg/C23" from the repo

```
+ sudo snap install curl
snap "curl" is already installed, see 'snap help refresh'
[Pipeline] sh
+ which curl
/usr/bin/curl
[Pipeline] sh
+ /snap/bin/curl --version
[Pipeline] withCredentials
Masking supported pattern matches of $AWS_ACCESS_KEY_ID or $AWS_SECRET_ACCESS_KEY
[Pipeline] {
[Pipeline] sh
+ AWS_DEFAULT_REGION=us-west-2 aws lambda get-function-url-config --function-name release_writer_staging

[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] withCredentials
Masking supported pattern matches of $AWS_ACCESS_KEY_ID or $AWS_SECRET_ACCESS_KEY
[Pipeline] {
[Pipeline] sh
+ /snap/bin/curl --aws-sigv4 aws:amz:us-west-2:lambda -w \n%{http_code} --user ****:**** https://md5rfjquft42xfe3kvqmse7xsy0wjdkw.lambda-url.us-west-2.on.aws/ -H Content-Type: application/json -d {"route":"builds","data":{"project_id":"secondary-watcher-pg","build_id":"secondary-watcher-pg:C23","release_id":"C23","purpose":"initial","git_tag":"soql-server-pg/C23"}}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   173  100     2  100   171     46   3996 --:--:-- --:--:-- --:--:--  4119
```

**Comment by sarah-salinas** [comment]
## Implementation

- [] Change RMS environment from staging to production
- [] Wait until Release-Build changes are merged

**Review by rjmac — APPROVED** [review]

File: Jenkinsfile
Comment (ds2643) [inline Jenkinsfile:93]: I generally don't like to comment on formatting, but I think newlines would preserve the git history nicely for posterity in cases where number of arguments > 3:
```groovy
createBuild(
  releaseName: params.RELEASE_NAME,
  serviceName: service_server,
  buildID: env.DOCKER_TAG,
  gitTag: env.GIT_TAG,
  rmsEnvironment: rmsSupportedEnvironment.staging
)
```
This formatting also makes the list of arguments easier to read.

**Review by ds2643 — COMMENTED** [review]
This invocation of the `pipelines` code is fairly non-invasive. I'd like feedback on the results from my supporting updates to your `createBuild` wrapper in Pipelines before explicitly approving, but I have little reservation about doing so, provided testing goes well.

Good work.

File: Jenkinsfile
Comment (sarah-salinas) [reply Jenkinsfile]: Updated formatting

**Comment by sarah-salinas** [comment]
## Retesting

After changes to the pipeline on this PR branch: https://github.com/socrata/pipelines/pull/149

https://jenkins-build.socrata.com/job/soql-postgres-adapter-branch/76/console
```
Post stage
[Pipeline] script
[Pipeline] {
[Pipeline] echo
In the createBuild method
[Pipeline] sh
+ sudo snap install curl
snap "curl" is already installed, see 'snap help refresh'
[Pipeline] sh
+ which curl
/usr/bin/curl
[Pipeline] sh

+ /snap/bin/curl --version
[Pipeline] withCredentials
Masking supported pattern matches of $AWS_ACCESS_KEY_ID or $AWS_SECRET_ACCESS_KEY
[Pipeline] {
[Pipeline] sh
+ AWS_DEFAULT_REGION=us-west-2 aws lambda get-function-url-config --function-name release_writer_staging
[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] withCredentials
Masking supported pattern matches of $AWS_ACCESS_KEY_ID or $AWS_SECRET_ACCESS_KEY
[Pipeline] {
[Pipeline] sh
+ /snap/bin/curl --aws-sigv4 aws:amz:us-west-2:lambda -w \n%{http_code} --user ****:**** https://md5rfjquft42xfe3kvqmse7xsy0wjdkw.lambda-url.us-west-2.on.aws/ -H Content-Type: application/json -d {"route":"builds","data":{"project_id":"secondary-watcher-pg","build_id":"secondary-watcher-pg:D23","release_id":"D23","git_tag":"soql-server-pg/D23"}}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   153  100     2  100   151     20   1543 --:--:-- --:--:-- --:--:--  1561
100   153  100     2  100   151     20   1541 --:--:-- --:--:-- --:--:--  1561
[Pipeline] readJSON
[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] }
[Pipeline] // script
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
```

File: Jenkinsfile
Comment (ds2643) [inline Jenkinsfile:93]: What happens in the case of a hotfix?

File: Jenkinsfile
Comment (ds2643) [inline Jenkinsfile:95]: I'm very skeptical of coercing these values into strings via interpolation. Aren't the associated values already optional strings (i.e., string or null)? I'm very afraid that we'll end up with `"null"` in the database.

**Review by ds2643 — CHANGES_REQUESTED** [review]
I don't think its safe to coerce the `buildInfo` values to strings via interpolation because we could capture `null` as a string. Not all columns are validated, so we could end up with junk data in the database.

File: Jenkinsfile
Comment (sarah-salinas) [reply Jenkinsfile:93]: We don't currently support hotfixes for soql-postgres-adapter

**Comment by sarah-salinas** [comment]
## Retested

https://jenkins-build.socrata.com/job/soql-postgres-adapter-branch/80/console - 🟢 
```
+ /snap/bin/curl --aws-sigv4 aws:amz:us-west-2:lambda -w \n%{http_code} --user ****:**** https://md5rfjquft42xfe3kvqmse7xsy0wjdkw.lambda-url.us-west-2.on.aws/ -H Content-Type: application/json -d {"route":"builds","data":{"project_id":"secondary-watcher-pg","build_id":"secondary-watcher-pg:B23","release_id":"B23","git_tag":"soql-server-pg/B23"}}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   153  100     2  100   151     53   4060 --:--:-- --:--:-- --:--:--  4135
```

**Comment by sarah-salinas** [comment]
## Retesting

https://jenkins-build.socrata.com/job/soql-postgres-adapter-branch/81/console
soql-server-pg
```
+ sudo snap install curl

curl 8.1.2 from Wouter van Bommel (woutervb) installed
[Pipeline] sh
+ which curl
/usr/bin/curl
[Pipeline] sh
+ /snap/bin/curl --version

[Pipeline] withCredentials
Masking supported pattern matches of $AWS_ACCESS_KEY_ID or $AWS_SECRET_ACCESS_KEY
[Pipeline] {
[Pipeline] sh
+ AWS_DEFAULT_REGION=us-west-2 aws lambda get-function-url-config --function-name release_writer_staging

[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] withCredentials
Masking supported pattern matches of $AWS_ACCESS_KEY_ID or $AWS_SECRET_ACCESS_KEY
[Pipeline] {
[Pipeline] sh
+ /snap/bin/curl --aws-sigv4 aws:amz:us-west-2:lambda -w \n%{http_code} --user ****:**** https://md5rfjquft42xfe3kvqmse7xsy0wjdkw.lambda-url.us-west-2.on.aws/ -H Content-Type: application/json -d {"route":"builds","data":{"project_id":"soql-server-pg","build_id":"soql-server-pg:F23","release_id":"F23","git_tag":"soql-server-pg/F23"}}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   139    0     0  100   139      0    115  0:00:01  0:00:01 --:--:--   115
100   141  100     2  100   139      1     69  0:00:02  0:00:01  0:00:01    70
100   141  100     2  100   139      1     69  0:00:02  0:00:01  0:00:01    70
```

secondary-watcher-pg
```

+ sudo snap install curl
snap "curl" is already installed, see 'snap help refresh'
[Pipeline] sh
+ which curl
/usr/bin/curl
[Pipeline] sh
+ /snap/bin/curl --version
[Pipeline] withCredentials
Masking supported pattern matches of $AWS_ACCESS_KEY_ID or $AWS_SECRET_ACCESS_KEY
[Pipeline] {
[Pipeline] sh

+ AWS_DEFAULT_REGION=us-west-2 aws lambda get-function-url-config --function-name release_writer_staging
[Pipeline] }
[Pipeline] // withCredentials
[Pipeline] withCredentials
Masking supported pattern matches of $AWS_ACCESS_KEY_ID or $AWS_SECRET_ACCESS_KEY
[Pipeline] {
[Pipeline] sh

+ /snap/bin/curl --aws-sigv4 aws:amz:us-west-2:lambda -w \n%{http_code} --user ****:**** https://md5rfjquft42xfe3kvqmse7xsy0wjdkw.lambda-url.us-west-2.on.aws/ -H Content-Type: application/json -d {"route":"builds","data":{"project_id":"secondary-watcher-pg","build_id":"secondary-watcher-pg:F23","release_id":"F23","git_tag":"soql-server-pg/F23"}}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   153  100     2  100   151     42   3175 --:--:-- --:--:-- --:--:--  3255
```

**Review by rjmac — APPROVED** [review]

File: Jenkinsfile
Comment (ds2643) [inline Jenkinsfile:52]: I'm not a huge fan of passing around global state with environment variables. Is there some reason why a normal global variable won't suffice? Perhaps I'm misreading the code.

File: Jenkinsfile
Comment (ds2643) [inline Jenkinsfile:8]: The naming is preexisting, but I think `server_service_name` better aligns with our conventions elsewhere. I won't worry about updating given that such naming conventions aren't relevant to the PR's purpose.

**Review by ds2643 — APPROVED** [review]
Thanks for taking my feedback into account. The integration looks clean.

File: Jenkinsfile
Comment (sarah-salinas) [reply Jenkinsfile:52]: Jenkins has some weird rules about variables and their scope based on when they are defined.  I would have to define gitTag outside of the pipeline in order for me to set it in one stage and use it in another.  Since it doesn't have a default value, I'd rather define it as an environment variable which will persist through all stages if it's defined in the script section and not the environment section.

**Pull request merged**
